sys/net/routing/rpl/rpl_dodag.c

```diff
@@ -235,16 +231,16 @@ void rpl_delete_worst_parent(void)
 
 }
 
-void rpl_delete_all_parents(void)
+void rpl_delete_all_parents(rpl_dodag_t *dodag)
 {
-    rpl_dodag_t *my_dodag = rpl_get_my_dodag();
-
-    if (my_dodag != NULL) {
-        my_dodag->my_preferred_parent = NULL;
-    }
 
+    dodag->my_preferred_parent = NULL;
     for (int i = 0; i < RPL_MAX_PARENTS; i++) {
-        memset(&parents[i], 0, sizeof(parents[i]));
+        if (dodag->instance->id == parents[i].dodag->instance->id &&
```

Comment:
@OlegHahm @cgundogan found it, 
@cgundogan could you add a check if the specific parent has a DODAG set, for instance:
``` C
if (parents[i].dodag && dodag->instance->id == parents[i].dodag->instance->id &&
```
Now iterating over the parents without a set DODAG will crash on access `->instance...`